### PR TITLE
Cleanup v2 with some updates

### PR DESCRIPTION
**Pull request by yashrsharma44**
Cleanup some leftover errchecks and updated bunch of dependencies required
Also renamed tocken to token for `providers/tokenbucket`

Signed-off-by: Yash Sharma <yashrsharma44@gmail.com>

cc @bwplotka @johanbrandhorst

### CODE REVIEW

**Review by johanbrandhorst — APPROVED** [review]

File: interceptors/auth/examples_test.go [hunk 1]
```diff
@@ -51,14 +51,18 @@ func Example_serverConfig() {
...
+	pb.UnimplementedGreeterServer
+}
 
 // SayHello only can be called by client when authenticated by exampleAuthFunc
 func (g gRPCserverAuthenticated) SayHello(ctx context.Context, request *pb.HelloRequest) (*pb.HelloReply, error) {
 	return &pb.HelloReply{Message: "pong authenticated"}, nil
 }
 
-type gRPCserverUnauthenticated struct{}
+type gRPCserverUnauthenticated struct {
```
Comment (bwplotka) [inline interceptors/auth/examples_test.go:63]:
```suggestion
type gRPCServerUnauthenticated struct {
```

File: interceptors/auth/examples_test.go [hunk 2]
```diff
@@ -51,14 +51,18 @@ func Example_serverConfig() {
 	)
 }
 
-type gRPCserverAuthenticated struct{}
+type gRPCserverAuthenticated struct {
```
Comment (bwplotka) [inline interceptors/auth/examples_test.go:54]:
```suggestion
type gRPCServerAuthenticated struct {
```

**Review by bwplotka — APPROVED** [review]
Amazing, thanks!

**Review by bwplotka — APPROVED** [review]
Amazing, thank you!

**Pull request merged**